allchblk.c

```diff
@@ -1005,6 +1005,12 @@ STATIC struct hblk *GC_allochblk_nth(size_t lb_adjusted, int k,
     return hbp;
 }
 
+# if defined(VALGRIND_TRACKING)
+    GC_API void GC_CALL GC_free_profiler_hook(void * p) {
+        UNUSED_ARG(p);
```

Comment:
I guess we should use GC_noop1((word)p) instead. Otherwise a compiler with -O3 option could remove such calls from the code.

Reply:
But probably it is ok to leave UNUSED if no in line is added.

Reply:
That makes sense, I've fixed this now :)